src/Controller/ResultController.php

```diff
@@ -43,26 +44,19 @@ public function __construct(RendererInterface $renderer, ResultService $resultSe
      * @param array $requestAttributes
      * @return Message|MessageInterface
      */
-    public function showResults(Request $request, array $requestAttributes)
+    public function showResults(Request $request, array $requestAttributes): MessageInterface
     {
         $redirectToLogin = $this->verifySessionUserName($this->resultService->getSession());
```

Comment:
Consider extracting authorization to a higher level. In the dispatcher for example.  There is an approach in the document provided by Andra a few weeks ago.